HelpSource/Classes/String.schelp

```diff
@@ -372,16 +372,19 @@ code::
 ::
 
 method::findRegexp
-POSIX regular expression search.
+POSIX regular expression search. This method searches exhaustively for all possible matches and collects them into an array of pairs, in the format code::[character index, matching string]::. As in the regular expression standard, code::*:: and code::+:: are greedy; the match list will include the largest possible distinct matches. Note, though, that parentheses for grouping will produce a separate result: code::"aaa".findRegexp("(a+)");:: appears to produce duplicated results code::[ [ 0, aaa ], [ 0, aaa ] ]::, but this is because the first match is for the parentheses and the second is for code::a+::.
```

Comment:
fairly certain it's actually Perl flavored regex. https://www.boost.org/doc/libs/1_69_0/libs/regex/doc/html/boost_regex/syntax.html

>  There are three main syntax options available, depending upon how you construct the regular expression object:
>  - Perl (this is the default behavior). 

you can double check this but otherwise I don't think a match like `"aaa".findRegexp("aa*?")` would work. also, this does not search 'exhaustively for all possible matches', see for example `"aaa".findRegexp("aa")`. https://www.boost.org/doc/libs/1_69_0/libs/regex/doc/html/boost_regex/syntax/leftmost_longest_rule.html is relevant.

it seems the terminology for unescaped parens in this regex flavor is 'marked sub-expression', but i've also seen 'capturing group'. a more illustrative example to show how results are returned for marked sub-expressions might be `"_ab_".findRegexp("(a)(b)")`. (or just annotate the example below for `"foobar".findRegexp("(o*)(bar)");`

it would probably be a good idea to link directly to the documentation for this syntax. or copy the documentation here with appropriate attribution. I would include at least https://www.boost.org/doc/libs/1_69_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html, possibly also that leftmost-longest-rule document as cryptic as it is.